src/js/engine.js

```diff
@@ -2562,7 +2562,7 @@ function processInput(dir,dontDoWin,dontModify) {
 		}
 
 		
-        bannedGroup = [];
+        level.bannedGroup = [];
```

Comment:
Why are you putting bannedGroup in level here?  It looks like it's just kept/passed around as a local/transitory object.

Reply:
Oh right, there's no var here, so it's global.  I'll just accept your pull request and modify it to be a var.